protocol/contracts/zaps/ZapBase.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.8.4;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IWeth.sol";
+
+abstract contract ZapBase is Ownable {
+  using SafeERC20 for IERC20;
+
+  bool public paused;
+  address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+  uint256 internal constant DEADLINE = 0xf000000000000000000000000000000000000000000000000000000000000000;
+
+  // fromToken => swapTarget (per curve, univ2 and balancer) approval status
+  mapping(address => mapping(address => bool)) public approvedTargets;
+
+  event SetContractState(bool paused);
+
+  receive() external payable {
+    require(msg.sender != tx.origin, "Do not send ETH directly");
```

Comment:
nit: prefix error messages with contract name (can use ZB or similar abbreviation if we're attempting to stick < 32 chars)